### PR TITLE
Fourth column

### PR DESCRIPTION
**Pull request by hyx3179**
Added option for KS to be displayed in the fourth column

### CODE REVIEW

**Comment by hyx3179** [comment]
#493

File: packages/kitten-scientists/source/settings/EngineSettings.ts [hunk 1]
```diff
@@ -52,6 +56,7 @@ export class EngineSettings extends Setting {
       this.states.load(settings.states);
       this.language.load(settings.language);
     }
+    this.ksColumn.enabled = localStorage["ks.Column"] === "#ksColumn";
```
Comment (oliversalzburg) [inline packages/kitten-scientists/source/settings/EngineSettings.ts:59]: Don't access localStorage in the loader. Use the .load method of the Setting to copy the state.

The ID of the target column does not need to be stored. You only need the true or false information.

File: packages/kitten-scientists/source/ui/InternalsUi.ts [hunk 2]
```diff
@@ -41,6 +42,16 @@ export class InternalsUi extends SettingsPanel<EngineSettings> {
                 },
               }),
             ),
+            new SettingListItem(host, host.engine.i18n("ui.ksColumn"), settings.ksColumn, {
+              onCheck: () => {
+                localStorage.setItem("ks.Column", "#ksColumn");
```
Comment (oliversalzburg) [inline packages/kitten-scientists/source/ui/InternalsUi.ts:47]: You should not need to remember anything in localStorage here. The information is already preserved in the settings.

File: packages/kitten-scientists/source/ui/UserInterface.ts [hunk 3]
```diff
@@ -117,7 +117,15 @@ export class UserInterface extends UiComponent {
     $("#clearLog").prepend(this.showActivity);
 
     // Add Kitten Scientists above the game log.
-    const right = $("#rightColumn");
+    const id = localStorage["ks.Column"] as string | undefined;
```
Comment (oliversalzburg) [inline packages/kitten-scientists/source/ui/UserInterface.ts:120]: Read the state from the settings. Then pick either #rightColumn or #ksColumn.

File: packages/kitten-scientists/source/ui/UserInterface.ts [hunk 4]
```diff
@@ -117,7 +117,15 @@ export class UserInterface extends UiComponent {
...
+    if (id === "#ksColumn" && !$("#ksColumn").length) {
+      $("#rightColumn").after(
+        '<div id="ksColumn" class="column"><br><br><br><br><br><br><br><br><br><br></div>',
```
Comment (oliversalzburg) [inline packages/kitten-scientists/source/ui/UserInterface.ts:123]: Can the br tags be replaced with CSS?

**Review by oliversalzburg — CHANGES_REQUESTED** [review]

File: packages/kitten-scientists/source/ui/UserInterface.ts (hunk 3 above)
Comment (hyx3179) [reply packages/kitten-scientists/source/ui/UserInterface.ts:120]: image
When entering the web page, UI drawing is executed before loading the settings, and the correct settings cannot be executed.

Comment (hyx3179) [reply packages/kitten-scientists/source/ui/UserInterface.ts:120]: Maybe redraw the UI after loading the settings? let me try

Comment (oliversalzburg) [reply packages/kitten-scientists/source/ui/UserInterface.ts]: Maybe this is similar to the language setting. Check out if the same approach should be used here:

https://github.com/kitten-science/kitten-scientists/blob/4d7f05981a93783f6b3b842e515012b49ba13b54/packages/kitten-scientists/source/Engine.ts#L149-L151

**Review by oliversalzburg — APPROVED** [review]
lgtm

**Pull request merged**